### PR TITLE
Add Jamie's Passworks post.

### PR DESCRIPTION
**Pull request by krashanoff**
Jamie just finished up her passworks post. This PR is to provide her a spot to preview it on the site and to serve as grounds for any other tweaks that might need to be made.

Only notable change to the website's code is the addition of the `.post-img` class, which simply sets the max-width of images to 96% with 2% padding on all sides.

Preview: https://deploy-preview-89--acm-teach-la-website.netlify.app/blog/dev/2020/09/23/the-making-of-passworks/

### CODE REVIEW

**Comment by malsf21** [comment]
Overall, I think this looks good. I'll see if I can cherry-pick in the syntax highlighting css from the internal docs merge; if not, we'll deal with that merge conflict later.

File: _posts/2020-09-22-the-making-of-passworks.md [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+### The Lesson Framework
+
+Probably one of the biggest decisions during the building of the app was how I would store the lessons/activities in the app. As I was fleshing out some details in the activities, I noticed that each "slide" of a lesson would consist largely of the same elements, in particular, some descriptive text and some content on the phone (see poorly drawn example below).
+
+![sample text on a phone screen]({{site.baseurl}}/img/posts/passworks/layout.png){:class='post-img'}
+
+In addition, there were many instances where two different activities would be able to reuse previous components. For example, the first two activities both feature a password input and password generator. I realized that I could create a declarative framework, where each lesson is defined by an array of JavaScript objects. For example, the block of code below represents a single slide in a lesson:
+
+```js
```
Comment (malsf21) [inline _posts/2020-09-22-the-making-of-passworks.md:65]: does JSX work here? highlithing is weird

File: _posts/2020-09-22-the-making-of-passworks.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+## Components vs. Hooks
+
+This is not so much a bug as it was something I noticed or learned throughout the development process, especially working with the `react-countup` module. Initially, I had a lot of difficulties implementing the timer portion of the password-guessing slide, since both of them used the component form of the `react-countup` module and I didn't know how to connect them to the same start button. I believe a lot of this can be attributed to confusion with components and the fact that composing two CountUp components is non-trivial. I ended up switching from the component implementation of `CountUp` to the `useCountUp` hook, and things became much more simple.
+
+[This commit](https://github.com/uclaacm/passworks/commit/eaaf2527cae3cdb232a93335640a248e57f262a8) contains most of the work I had to do while transitioning from components to hooks, and provides a decent illustration of the gains from fully switching to hooks. In particular, compare the following code snippets: the first using the `CountUp` component and the second using the equivalent hook:
+
+```js
```
Comment (malsf21) [inline _posts/2020-09-22-the-making-of-passworks.md:189]: jsx?

File: _posts/2020-09-22-the-making-of-passworks.md [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+            <span ref={ref2} />
+            <Button onClick={() => { start1(); start2(); }}>Start</Button>
+          </Box>
+        )
+      </CountUp>
+    )}
+  </CountUp>
+);
+```
+```js
```
Comment (malsf21) [inline _posts/2020-09-22-the-making-of-passworks.md:212]: jsx?

File: _posts/2020-09-22-the-making-of-passworks.md [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+
+---
+
+### Regular Expressions
+
+I used regular expressions to validate user inputs by writing callbacks that tested the user's input against hard-coded regular expressions. They weren't particularly advanced, but it was a reminder of how cool and powerful regular expressions can be (and provided me some much-needed practice).
+
+---
+
+<a id="counting-up-to-strings"></a>
```
Comment (malsf21) [inline _posts/2020-09-22-the-making-of-passworks.md:259]: what is this doing here?

Comment (krashanoff) [reply _posts/2020-09-22-the-making-of-passworks.md:259]: Huh. This was just in the original doc.

**Review by malsf21 — APPROVED** [review]
Awesome, thanks for all the work @krashanoff and @jamieliu386, I think this LGTM! We'll deal with syntax highlighting another day.

**Pull request merged**